shell/platform/embedder/embedder_semantics_update.cc

```diff
@@ -175,4 +193,61 @@ void EmbedderSemanticsUpdate2::AddAction(
   });
 }
 
+std::pair<size_t, const FlutterStringAttribute**>
+EmbedderSemanticsUpdate2::CreateStringAttributes(
+    const StringAttributes& attributes) {
+  // Minimize allocations if attributes are empty.
+  if (attributes.empty()) {
+    return std::make_pair(0, nullptr);
+  }
+
+  auto result = std::make_unique<std::vector<const FlutterStringAttribute*>>();
+  result->reserve(attributes.size());
+
+  for (const auto& attribute : attributes) {
+    auto embedder_attribute = std::make_unique<FlutterStringAttribute>();
+    embedder_attribute->struct_size = sizeof(FlutterStringAttribute);
+    embedder_attribute->start = attribute->start;
+    embedder_attribute->end = attribute->end;
+
+    switch (attribute->type) {
+      case StringAttributeType::kLocale: {
+        std::shared_ptr<flutter::LocaleStringAttribute> locale_attribute =
+            std::static_pointer_cast<flutter::LocaleStringAttribute>(attribute);
+
+        auto embedder_locale = std::make_unique<FlutterLocaleStringAttribute>();
+        embedder_locale->struct_size = sizeof(FlutterLocaleStringAttribute);
+        embedder_locale->locale = locale_attribute->locale.c_str();
+        locale_attributes_.push_back(std::move(embedder_locale));
```

Comment:
Can we write this as:
```cc
locale_attributes_.emplace_back({fields, in, declaration, order});
```
That said, I'd have to look it up; struct initializer support without an explicit constructor may only have been added in C++20. I feel like I may have had the same discussion with knopp recently.

Similar below.

Reply:
Yes, our compiler doesn't compile this without an explicit constructor on the struct :(